src/Exceptions/InvalidPayloadException.php

```diff
@@ -2,15 +2,36 @@
 
 namespace Cubex\ApiFoundation\Exceptions;
 
+use Throwable;
+
 class InvalidPayloadException extends \Exception
 {
+  /** @var string */
   protected $message = 'Invalid payload';
+
+  /** @var int */
   protected $code = 400;
 
-  public static function withType($type): static
+  /**
+   * @param string         $type
+   * @param string         $message
+   * @param int            $code
+   * @param Throwable|null $previous
+   */
+  public function __construct(string $type, string $message = "", int $code = 0, ?Throwable $previous = null)
```

Comment:
This doesn't follow the constructor for Exception

Reply:
As we are looking to add a type I don't see why we want it to follow the exception constructor.